packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -401,27 +416,54 @@ export default declare((api, opts) => {
               t.assignmentExpression("=", left, t.cloneNode(temp)),
             ),
           );
+        } else {
+          // for (var {a, ...b} of []) {}
+          const pattern = left.declarations[0].id;
 
-          return;
+          const key = scope.generateUidIdentifier("ref");
+          node.left = t.variableDeclaration(left.kind, [
+            t.variableDeclarator(key, null),
+          ]);
+
+          path.ensureBlock();
+
+          node.body.body.unshift(
```

Comment:
`node.body` could be an `EmptyStatement`
```js
for (const foo of []);
```

Reply:
`path.ensureBlock` has ensure that it will be a block statement.